### DISCUSSION TITLE
undefined req.cookies

### DISCUSSION THREAD

**Discussion by naissa12** [vercel/next.js#11138] [Help]
I'm trying to make a request to the express/nodejs backend using nextjs

in pages/reader.js, I have

```
Reader.getInitialProps = async ({query}) => {
    const res = await fetch('http://localhost:3000/api/books/reader/' + query.id);
    const json = await res.json();
    return {book: json}
};
```
Unfortunately, that overwrites the cookies stored in the request object on the backend. When I do a `console.dir(req.cookies)` in the backend node js, express code, I get undefined in book.js where the reader code is.

How can I fetch without overwriting the request object in the express backend?

**Comment by jamesmosier** [comment]
Have you tried 

```js
const res = await fetch('http://localhost:3000/api/books/reader/' + query.id, { 
  credentials: "include" 
})
```

**Reply by NamDo8467** [reply]
This solved my problem, thanks. And for the record, for people who use axios, it should be "withCredentials:  true"
